libr/util/sys.c

```diff
@@ -122,6 +121,34 @@ R_API int r_sys_truncate(const char *file, int sz) {
 
 R_API RList *r_sys_dir(const char *path) {
 	RList *list = NULL;
+#if __WINDOWS__ && !defined(__CYGWIN__)
+	HANDLE fh;
+	WIN32_FIND_DATAW entry;
+	wchar_t dir[MAX_PATH];
+	wchar_t *wcpath;
+	char *cfname;
+
+	fh = r_sandbox_opendir (path, &entry, dir, wcpath);
+	if (fh == INVALID_HANDLE_VALUE) {
+		//IFDGB eprintf ("Cannot open directory %ls\n", wcpath);
+		if (wcpath) {
+			free (wcpath);
+		}
+		return list;
+	}
+	list = r_list_new ();
```

Comment:
use r_list_newf instead of later assigning the ->free pointer on every iteration